File: frameworks/java/spring-boot/Client/Kiota/client/kiota/models/NewItem.java
package client.kiota.models;

import com.microsoft.kiota.serialization.AdditionalDataHolder;
import com.microsoft.kiota.serialization.Parsable;
import com.microsoft.kiota.serialization.ParseNode;
import com.microsoft.kiota.serialization.SerializationWriter;
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;
@jakarta.annotation.Generated("com.microsoft.kiota")
public class NewItem implements AdditionalDataHolder, Parsable {
    /**
     * Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     */
    private Map<String, Object> additionalData;
    /**
     * The category property
     */
    private String category;
    /**
     * The inStock property
     */
    private Boolean inStock;
    /**
     * The name property
     */
    private String name;
    /**
     * The priceCents property
     */
    private Integer priceCents;
    /**
     * Instantiates a new {@link NewItem} and sets the default values.
     */
    public NewItem() {
        this.setAdditionalData(new HashMap<>());
    }
    /**
     * Creates a new instance of the appropriate class based on discriminator value
     * @param parseNode The parse node to use to read the discriminator value and create the object
     * @return a {@link NewItem}
     */
    @jakarta.annotation.Nonnull
    public static NewItem createFromDiscriminatorValue(@jakarta.annotation.Nonnull final ParseNode parseNode) {
        Objects.requireNonNull(parseNode);
        return new NewItem();
    }
    /**
     * Gets the AdditionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @return a {@link Map<String, Object>}
     */
    @jakarta.annotation.Nonnull
    public Map<String, Object> getAdditionalData() {
        return this.additionalData;
    }
    /**
     * Gets the category property value. The category property
     * @return a {@link String}
     */
    @jakarta.annotation.Nullable
    public String getCategory() {
        return this.category;
    }
    /**
     * The deserialization information for the current model
     * @return a {@link Map<String, java.util.function.Consumer<ParseNode>>}
     */
    @jakarta.annotation.Nonnull
    public Map<String, java.util.function.Consumer<ParseNode>> getFieldDeserializers() {
        final HashMap<String, java.util.function.Consumer<ParseNode>> deserializerMap = new HashMap<String, java.util.function.Consumer<ParseNode>>(4);
        deserializerMap.put("category", (n) -> { this.setCategory(n.getStringValue()); });
        deserializerMap.put("inStock", (n) -> { this.setInStock(n.getBooleanValue()); });
        deserializerMap.put("name", (n) -> { this.setName(n.getStringValue()); });
        deserializerMap.put("priceCents", (n) -> { this.setPriceCents(n.getIntegerValue()); });
        return deserializerMap;
    }
    /**
     * Gets the inStock property value. The inStock property
     * @return a {@link Boolean}
     */
    @jakarta.annotation.Nullable
    public Boolean getInStock() {
        return this.inStock;
    }
    /**
     * Gets the name property value. The name property
     * @return a {@link String}
     */
    @jakarta.annotation.Nullable
    public String getName() {
        return this.name;
    }
    /**
     * Gets the priceCents property value. The priceCents property
     * @return a {@link Integer}
     */
    @jakarta.annotation.Nullable
    public Integer getPriceCents() {
        return this.priceCents;
    }
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    public void serialize(@jakarta.annotation.Nonnull final SerializationWriter writer) {
        Objects.requireNonNull(writer);
        writer.writeStringValue("category", this.getCategory());
        writer.writeBooleanValue("inStock", this.getInStock());
        writer.writeStringValue("name", this.getName());
        writer.writeIntegerValue("priceCents", this.getPriceCents());
        writer.writeAdditionalData(this.getAdditionalData());
    }
    /**
     * Sets the AdditionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param value Value to set for the AdditionalData property.
     */
    public void setAdditionalData(@jakarta.annotation.Nullable final Map<String, Object> value) {
        this.additionalData = value;
    }
    /**
     * Sets the category property value. The category property
     * @param value Value to set for the category property.
     */
    public void setCategory(@jakarta.annotation.Nullable final String value) {
        this.category = value;
    }
    /**
     * Sets the inStock property value. The inStock property
     * @param value Value to set for the inStock property.
     */
    public void setInStock(@jakarta.annotation.Nullable final Boolean value) {
        this.inStock = value;
    }
    /**
     * Sets the name property value. The name property
     * @param value Value to set for the name property.
     */
    public void setName(@jakarta.annotation.Nullable final String value) {
        this.name = value;
    }
    /**
     * Sets the priceCents property value. The priceCents property
     * @param value Value to set for the priceCents property.
     */
    public void setPriceCents(@jakarta.annotation.Nullable final Integer value) {
        this.priceCents = value;
    }
}

File: frameworks/python/sanic/Client/Kiota/models/new_item.py
from __future__ import annotations
from collections.abc import Callable
from dataclasses import dataclass, field
from kiota_abstractions.serialization import AdditionalDataHolder, Parsable, ParseNode, SerializationWriter
from typing import Any, Optional, TYPE_CHECKING, Union

@dataclass
class NewItem(AdditionalDataHolder, Parsable):
    """
    An item as a client creates or replaces one.
    """
    # Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
    additional_data: dict[str, Any] = field(default_factory=dict)

    # The category property
    category: Optional[str] = None
    # The inStock property
    in_stock: Optional[bool] = None
    # The name property
    name: Optional[str] = None
    # The priceCents property
    price_cents: Optional[int] = None
    
    @staticmethod
    def create_from_discriminator_value(parse_node: ParseNode) -> NewItem:
        """
        Creates a new instance of the appropriate class based on discriminator value
        param parse_node: The parse node to use to read the discriminator value and create the object
        Returns: NewItem
        """
        if parse_node is None:
            raise TypeError("parse_node cannot be null.")
        return NewItem()
    
    def get_field_deserializers(self,) -> dict[str, Callable[[ParseNode], None]]:
        """
        The deserialization information for the current model
        Returns: dict[str, Callable[[ParseNode], None]]
        """
        fields: dict[str, Callable[[Any], None]] = {
            "category": lambda n : setattr(self, 'category', n.get_str_value()),
            "inStock": lambda n : setattr(self, 'in_stock', n.get_bool_value()),
            "name": lambda n : setattr(self, 'name', n.get_str_value()),
            "priceCents": lambda n : setattr(self, 'price_cents', n.get_int_value()),
        }
        return fields
    
    def serialize(self,writer: SerializationWriter) -> None:
        """
        Serializes information the current object
        param writer: Serialization writer to use to serialize this model
        Returns: None
        """
        if writer is None:
            raise TypeError("writer cannot be null.")
        writer.write_str_value("category", self.category)
        writer.write_bool_value("inStock", self.in_stock)
        writer.write_str_value("name", self.name)
        writer.write_int_value("priceCents", self.price_cents)
        writer.write_additional_data_value(self.additional_data)
    

